test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs

```diff
@@ -34,6 +40,99 @@ public void Build_honors_UseStartup_with_string()
             Assert.Equal("MyStartupAssembly", host.Options.StartupAssembly);
         }
 
+
+        [Fact]
+        public void Build_RunsHostingStartupAssembliesIfSpecified()
+        {
+            var builder = CreateWebHostBuilder()
+                .UseSetting(WebHostDefaults.HostingStartupAssembliesKey, typeof(WebHostBuilderTests).GetTypeInfo().Assembly.FullName)
```

Comment:
You want a test with the short name? It's whatever `assembly.Load(new AssemblyName(name))` supports